source_code/main_mcu/src/GUI/gui_prompts.c

```diff
@@ -576,6 +576,37 @@ void gui_prompts_display_3line_information_on_screen_and_wait(confirmationText_t
     }
 }
 
+/*! \fn     gui_prompts_get_char_to_display(uint8_t const * current_pin, uint8_t index)
+*   \brief  Get the next PIN value to display or '*' if PIN should be hidden
+*   \param  current_pin         Array containing the pin
+*   \param  index               Index into current_pin array
+*   \return Character to display
+*/
+static cust_char_t gui_prompts_get_char_to_display(uint8_t const * current_pin, uint8_t index)
```

Comment:
out of curiosity, do you get a casting warning because of the const?

Reply:
no, you won't get a warning for that conversion (its an "upgrade" - non-const -> const). You should only get a warning if the variable is const and you pass it to a function taking a non-const.